### PR TITLE
En 869 profile type endpoints

### PR DESCRIPTION
**Pull request by eugeniosant**

### CODE REVIEW

**Comment by entando-service** [comment]
All is well

File: engine/src/main/java/org/entando/entando/aps/system/services/userprofile/UserProfileTypeService.java [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+    @Override
+    public UserProfileTypeDto updateUserProfileType(ProfileTypeDtoRequest request, BindingResult bindingResult) {
+        return super.updateEntityType(SystemConstants.USER_PROFILE_MANAGER, request, bindingResult);
+    }
+
+    @Override
+    protected IUserProfile createEntityType(IEntityManager entityManager, EntityTypeDtoRequest dto, BindingResult bindingResult) throws Throwable {
+        IUserProfile userProfile = super.createEntityType(entityManager, dto, bindingResult);
+        /*
```
Comment (joewhite101) [inline engine/src/main/java/org/entando/entando/aps/system/services/userprofile/UserProfileTypeService.java]: delete

File: engine/src/main/java/org/entando/entando/aps/system/services/userprofile/model/UserProfileTypeDto.java [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+    private String listModel;
+    private String defaultModel;
+     */
+    public UserProfileTypeDto() {
+        super();
+    }
+
+    public UserProfileTypeDto(IUserProfile userProfile, List<AttributeRole> roles) {
+        super(userProfile, roles);
+        /*
```
Comment (joewhite101) [inline engine/src/main/java/org/entando/entando/aps/system/services/userprofile/model/UserProfileTypeDto.java]: delete

File: engine/src/main/java/org/entando/entando/aps/system/services/userprofile/model/UserProfileTypeDto.java [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+        this.setDefaultModel(dataObject.getDefaultModel());
+        this.setViewPage(dataObject.getViewPage());
+         */
+        List<AttributeInterface> entityAttributes = userProfile.getAttributeList();
+        for (AttributeInterface attribute : entityAttributes) {
+            EntityAttributeDto attributeDto = new EntityAttributeDto(attribute, roles);
+            this.getAttributes().add(attributeDto);
+        }
+    }
+    /*
```
Comment (joewhite101) [inline engine/src/main/java/org/entando/entando/aps/system/services/userprofile/model/UserProfileTypeDto.java]: delete

File: engine/src/main/java/org/entando/entando/web/userprofile/ProfileTypeController.java [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+        return profileTypeValidator;
+    }
+
+    public void setProfileTypeValidator(ProfileTypeValidator profileTypeValidator) {
+        this.profileTypeValidator = profileTypeValidator;
+    }
+
+    @RestAccessControl(permission = Permission.SUPERUSER)
+    @RequestMapping(method = RequestMethod.GET, produces = MediaType.APPLICATION_JSON_VALUE)
+    public ResponseEntity<?> getUserProfileTypes(RestListRequest requestList) throws JsonProcessingException {
```
Comment (joewhite101) [inline engine/src/main/java/org/entando/entando/web/userprofile/ProfileTypeController.java]: Why is this <?> and not < UserProfileTypeDto>

File: engine/src/main/java/org/entando/entando/web/userprofile/validator/ProfileTypeValidator.java [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+/**
+ * @author E.Santoboni
+ */
+@Component
+public class ProfileTypeValidator extends EntityTypeValidator {
+
+    /*
```
Comment (joewhite101) [inline engine/src/main/java/org/entando/entando/web/userprofile/validator/ProfileTypeValidator.java]: Delete

File: engine/src/main/java/org/entando/entando/aps/system/services/entity/model/EntityManagerDto.java [hunk 6]
```diff
@@ -1,86 +1,87 @@
...
+        return entityTypes;
+    }
+
+    public void setEntityTypes(List<EntityTypeDto> entityTypes) {
+        this.entityTypes = entityTypes;
+    }
+
+}
+
+/*
```
Comment (joewhite101) [inline engine/src/main/java/org/entando/entando/aps/system/services/entity/model/EntityManagerDto.java]: Remove. Or move to test class

File: engine/src/main/java/org/entando/entando/aps/system/services/userprofile/UserProfileTypeService.java (hunk 1 above)
Comment (eugeniosant) [reply engine/src/main/java/org/entando/entando/aps/system/services/userprofile/UserProfileTypeService.java]: Deleted!

File: engine/src/main/java/org/entando/entando/aps/system/services/userprofile/model/UserProfileTypeDto.java (hunk 2 above)
Comment (eugeniosant) [reply engine/src/main/java/org/entando/entando/aps/system/services/userprofile/model/UserProfileTypeDto.java]: Done!

File: engine/src/main/java/org/entando/entando/aps/system/services/userprofile/model/UserProfileTypeDto.java (hunk 3 above)
Comment (eugeniosant) [reply engine/src/main/java/org/entando/entando/aps/system/services/userprofile/model/UserProfileTypeDto.java]: Done!

File: engine/src/main/java/org/entando/entando/web/userprofile/ProfileTypeController.java (hunk 4 above)
Comment (eugeniosant) [reply engine/src/main/java/org/entando/entando/web/userprofile/ProfileTypeController.java]: The <?> is org.entando.entando.web.common.model.RestResponse, the object that contains the main payload, errors and metadata. Fixed!

File: engine/src/main/java/org/entando/entando/aps/system/services/entity/model/EntityManagerDto.java (hunk 6 above)
Comment (eugeniosant) [reply engine/src/main/java/org/entando/entando/aps/system/services/entity/model/EntityManagerDto.java]: Removed

File: engine/src/main/java/org/entando/entando/web/userprofile/validator/ProfileTypeValidator.java (hunk 5 above)
Comment (eugeniosant) [reply engine/src/main/java/org/entando/entando/web/userprofile/validator/ProfileTypeValidator.java]: Deleted

**Comment by entando-service** [comment]
All is well

**Comment by entando-service** [comment]
All is well

**Pull request merged**